pkg/util/fedinformer/transform.go

```diff
@@ -0,0 +1,64 @@
+package fedinformer
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// StripUnusedFields is the transform function for shared informers,
+// it removes unused fields from objects before they are stored in the cache to save memory.
+func StripUnusedFields(obj interface{}) (interface{}, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		// shouldn't happen
+		return obj, nil
+	}
+	// ManagedFields is large and we never use it
+	accessor.SetManagedFields(nil)
+	return obj, nil
+}
+
+// StripUnusedFieldsForNode is the transform function for shared informers,
+// it removes unused fields from nodes before they are stored in the cache to save memory.
+func StripUnusedFieldsForNode(obj interface{}) (interface{}, error) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		// shouldn't happen
+		return obj, nil
+	}
+
+	aggregatedNode := &corev1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: node.Name,
+		},
+		Status: corev1.NodeStatus{
+			Allocatable: node.Status.Allocatable,
+			Conditions:  node.Status.Conditions,
+		},
+	}
+	return aggregatedNode, nil
+}
+
+// StripUnusedFieldsForPod is the transform function for shared informers,
+// it removes unused fields from pods before they are stored in the cache to save memory.
+func StripUnusedFieldsForPod(obj interface{}) (interface{}, error) {
```

Comment:
```suggestion
// PodTransformFunc is the dedicated transform function for Pod objects.
// It cleans up some parts of the object before it will be put into the controller
// cache to reduce memory usage.
//
// Note: this function removes most of the fields, please make sure your controller
// doesn't care for the removed fields, especially when use in shared informers.
func PodTransformFunc(obj interface{}) (interface{}, error) {
```
Rename function name and add more comments.